vector-stores/spring-ai-oracle-store/src/test/java/org/springframework/ai/vectorstore/OracleVectorStoreTests.java

```diff
@@ -0,0 +1,358 @@
+import org.springframework.ai.document.Document;
+import org.springframework.ai.embedding.EmbeddingModel;
+import org.springframework.ai.transformers.TransformersEmbeddingModel;
+import org.springframework.ai.vectorstore.filter.FilterExpressionTextParser;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Primary;
+import org.springframework.core.io.DefaultResourceLoader;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.util.CollectionUtils;
+import org.testcontainers.containers.ExecConfig;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.oracle.OracleContainer;
+
+import javax.sql.DataSource;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.ai.vectorstore.OracleVectorStore.DEFAULT_SEARCH_ACCURACY;
+
+@Testcontainers
+// @EnabledIfEnvironmentVariable(named = "OPENAI_API_KEY", matches = ".+")
+public class OracleVectorStoreTests {
+
+	@Container
+	static OracleContainer oracle23aiContainer = new OracleContainer("gvenzl/oracle-free:23-slim") {
+		protected void runInitScriptIfRequired() {
+			try {
+				final Map<String, String> envMap = new HashMap<>();
+				envMap.put("ORACLE_BASE", "/opt/oracle");
+				envMap.put("ORACLE_BASE_CONFIG", "/opt/oracle/product/23ai/dbhomeFree");
+				envMap.put("ORACLE_BASE_HOME", "/opt/oracle/product/23ai/dbhomeFree");
+				envMap.put("ORACLE_HOME", "/opt/oracle/product/23ai/dbhomeFree");
+				envMap.put("ORACLE_SID", "FREE");
+				envMap.put("NLS_LANG", ".AL32UTF8");
+
+				final ExecConfig ec = ExecConfig.builder().user("oracle").command(new String[] { "/bin/sh", "-c", """
+						/opt/oracle/product/23ai/dbhomeFree/bin/sqlplus -s / as sysdba <<EOF
+						       -- Exit on any errors
+						       WHENEVER SQLERROR EXIT SQL.SQLCODE
+
+						       -- Configure the size of the Vector Pool to 1 GiB.
+						       ALTER SYSTEM SET vector_memory_size=1G SCOPE=SPFILE;
+
+						       SHUTDOWN ABORT;
+						       STARTUP;
+
+						       exit;
+						EOF""" }).workDir("/opt/oracle").envVars(envMap).build();
+				execInContainer(ec);
```

Comment:
I wonder if this can be part of the Oracle image itself to avoid boilerplate code when testing. For example, adding a `VECTOR_STORE` env var with value `true` and then the container will ran the script for me. WDYT?

Reply:
That's a subject indeed. I'll ask Gerald about it. So far, the problem is that the parameter above must be set in order to use vector indexes. Then, the database has to be restarted. Because of this, the Connection (JDBC) can't be reestablished to restart the database. We have to connect directly from inside the container to restart it, as shown above.

Building another container image could do the trick, but it is overkill, IMHO. Again I'll ping him to find a better solution but in the meantime, this is the only way I found while using Testcontainers :(

Maybe the other way around is to mount a volume....